core/src/server/host_filtering.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! HTTP Host Header validation.
+
+use crate::Error;
+use http::uri::{InvalidUri, Uri};
+use route_recognizer::Router;
+use std::str::FromStr;
+
+/// Port pattern
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
+pub enum Port {
+	/// No port specified (default port)
+	Default,
+	/// Port specified as a wildcard pattern (*).
+	Any,
+	/// Fixed numeric port
+	Fixed(u16),
+}
+
+impl From<u16> for Port {
+	fn from(port: u16) -> Port {
+		Port::Fixed(port)
+	}
+}
+
+/// Represent the http URI scheme that is returned by the HTTP host header
+///
+/// <http-URI = "http:" "//" authority path-abempty [ "?" query ][ "#" fragment ]>
```

Comment:
`path-abempty`? :)

Reply:
Copied it from the spec, I assume it means non-empty :P